Added setFrameRate to change the frame rate of an open PS Eye camera

zelmon64 opened the pull request:
Added the ability to change the frame rate of an open camera by stopping it, changing the frame rate then restarting it. This was added for use with PSMoveService and makes it very easy to lower the frame rate in order to reduce the USB bandwidth required.

cboulay commented:
I'm worried that functions that auto-stop and auto-start the camera might break applications that are not expecting that. While I think a `setFrameRate` function is useful, I think it should return an error code if the camera is not stopped and it should be up to the client to stop the camera before calling the setFrameRate function.

And yes, we can modify PSMoveService's config tool to stop, set the frame rate, then start the camera again.

inspirit commented:
i would prefer to return error in case camera is working. but do not auto the stop/start...

zelmon64 commented:
Okay, I think I understand. I've removed the auto stop/start and changed it into a bool that returns false if the camera is streaming and if not it returns true and changes the frame rate. These changes can be seen [here](https://github.com/inspirit/PS3EYEDriver/commit/7e1c68335bf5583ef91899e4cc3461791f8471fe) and the coresponding changes to [PSMoveService here](https://github.com/zelmon64/PSMoveService/commit/d5d6113eb3830fa2cdc1882416d3361c6472beba). 

How do I proceed? Do I close this pull request and make a new one? Thanks for the advice and help.

cboulay commented:
You can do a `git rebase -i master` then in the little editor change all the but the first `pick` to `fixup`, and edit the commit message for the first commit to be more accurate. Then after you do that you can `git push -f`. This PR should auto-update to having just a single commit.

zelmon64 commented:
Well I obviously messed that up. After I did `git rebase -i master` and changed all but the fist pick to fixup it said `CONFLICT (content): Merge conflict in src/ps3eye.h`. So I did `git add src/ps3eye.h` then `git rebase --continue` but this happened at each fixup. 

Should I try again or just delete it all and start again since it's such a small addition?

cboulay commented:
Yeah rebasing can be tricky. At this point I would just start over. It'll be more trouble than it's worth to recover.

zelmon64 commented:
I thought I'd try one more commit and rebase just in case and it seems like it worked. Thanks again for the help and advice @cboulay

inspirit commented:
super! great thanx
